packages/dashboard/src/components/feature-graph/index.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect } from 'react';
+import { cluster, line, curveBundle, select, hierarchy, HierarchyPointNode } from 'd3';
+import { classes } from './styles.st.css';
+
+export interface Link {
+    source: string;
+    target: string;
+}
+
+interface Node {
+    name: string;
+    id: string;
+    group: number;
+    parent?: GraphNode;
+}
+
+export interface GraphData {
+    nodes: Node[];
+    links: Link[];
+}
+
+export interface IFeatureGraphProps {
+    selectedFeatureGraph: GraphData;
+}
+
+interface GraphNode {
+    id?: string;
+    name: string;
+    group: number;
+    children: Array<GraphNode | Node>;
+    parent?: GraphNode;
+}
+
+export const FeatureGraph = ({ selectedFeatureGraph }: IFeatureGraphProps) => {
```

Comment:
- Many comments are missing
- variable names are very unclear (e, f, d etc..)
- extract to methods with meaningful names

it's very hard to review this file

Reply:
Yeah I know, added some docs, will cleanup the names but in general d3 is a very ugly API that requires a lot of domain knowledge, It's a lot of existing code that I adapted I bit